### DISCUSSION TITLE
Revalidate route search params

### DISCUSSION THREAD

**Discussion by AndonMitev** [vercel/next.js#59351] [Help]
Hey guys any idea why after page is refresh next router is not respecting query params and is switching them?

There are 1 route with 2 different query params:

1. `http://localhost:3000/games?activeTab=trendy`
2. `http://localhost:3000/games?activeTab=all`

only `activeTab` is different, activeTab is managed by `Link` and query params are read trough `searchParams` as prop on page server component

when user is on `http://localhost:3000/games?activeTab=trendy` and refresh the page initially `trendy` is passed as query param but second later is changes to `all` **automatically** so basically after every refresh `all` is opened

Why is happening that what could cause such a behaviour using next @14.0.3

**Comment by dreyanz** [comment]
had the same issue. Were you able to fix it?
